### PR TITLE
feat(builder): add bundle gas stats to metrics

### LINKED ISSUE #379: builder: Metric for gas used throughput
**Describe the feature**
We should add a metric to the builder that allows us to graph gas used and gas limit throughputs in grafana.

### PR DESCRIPTION
**Pull request by alex-miao**
Closes #379 

## Proposed Changes

  - Log gas limit, gas used, and gas limit utilization for mined bundles

### CODE REVIEW

File: crates/builder/src/transaction_tracker.rs [hunk 1]
```diff
@@ -379,6 +387,14 @@ where
             TransactionTrackerMetrics::set_current_fees(None);
         }
     }
+
+    async fn get_mined_tx_gas_info(&self, tx_hash: H256) -> anyhow::Result<(U256, U256)> {
+        let (tx, tx_receipt) = tokio::try_join!(
+            self.provider.get_transaction(tx_hash),
```
Comment (alex-miao) [inline crates/builder/src/transaction_tracker.rs:393]: We technically already have info about the gas limit of the transaction, but it's a pretty messy refactor to pass it all the way through to where we can log it so choosing to just make this extra request instead.

**Review by 0xfourzerofour — APPROVED** [review]

File: crates/builder/src/bundle_sender.rs [hunk 2]
```diff
@@ -551,6 +557,14 @@ impl BuilderMetrics {
         metrics::increment_counter!("builder_bundle_fee_increases", "builder_index" => builder_index.to_string());
     }
 
+    fn set_bundle_gas_stats(gas_limit: U256, gas_used: U256) {
+        let gas_limit = gas_limit.as_u128() as f64;
+        let gas_used = gas_used.as_u128() as f64;
+        metrics::gauge!("builder_bundle_gas_limit", gas_limit);
+        metrics::gauge!("builder_bundle_gas_used", gas_used);
+        metrics::gauge!("builder_bundle_gas_utilization", gas_limit / gas_used);
```
Comment (dancoombs) [inline crates/builder/src/bundle_sender.rs:565]: Why did you decide to use a gauge here? These metrics might be hard to visualize.

I think these should likely be counters. `gas_limit` and `gas_used`. And then we can take the derivative across a minute for each to get gas rates, then divide the gas rates to get gas efficiency.

File: crates/builder/src/transaction_tracker.rs (hunk 1 above)
Comment (dancoombs) [reply crates/builder/src/transaction_tracker.rs:393]: This info is in the pool, yeah? Part of https://github.com/alchemyplatform/rundler/issues/362 will be to pipe more information about mined transactions from pool to builder, so lets improve this then.

File: crates/builder/src/bundle_sender.rs (hunk 2 above)
Comment (alex-miao) [reply crates/builder/src/bundle_sender.rs:565]: Oh interesting, I didn't consider that approach. Are we able to do this pointwise division between two different time series in Grafana?

Comment (dancoombs) [reply crates/builder/src/bundle_sender.rs:565]: We aren't, but the idea is to get a general sense of what our trend is there, so I think aggregated metrics are fine. For pointwise stuff, we can use the event logs.

**Review by dancoombs — APPROVED** [review]

**Pull request merged**